src/components/dashboard/FaceRecognition/FRError.js

```diff
@@ -17,39 +16,45 @@ if (Platform.OS === 'web') {
   Image.prefetch(Oops)
 }
 
-const FRError = props => {
-  const { styles } = props
-  const store = GDStore.useStore()
-  const { fullName } = store.get('profile')
+const FRError = ({ styles, screenProps }) => {
+  const { screenState } = screenProps
+  let { isValid, error: reason } = screenState
 
-  const isValid = get(props, 'screenProps.screenState.isValid', undefined)
-
-  let reason = get(props, 'screenProps.screenState.error', '')
-  if (reason instanceof Error || reason.message) {
+  if (reason instanceof Error || reason.message || reason.error) {
     if (reason.name === 'NotAllowedError') {
       reason = `Looks like GoodDollar doesn't have access to your camera. Please provide access and try again`
     } else {
-      reason = reason.message
+      reason = reason.message || reason.error
     }
   }
 
-  log.debug({ props, reason })
+  log.debug({ styles, screenProps, reason })
 
   //is the error mesage something we want to show to the user? currently only camera related
   const isRelevantError = reason.match(/camera/i) || reason === 'Permission denied'
   let error = isRelevantError
     ? reason
     : "You see, it's not that easy\n to capture your beauty :)\nSo, let's give it another shot..."
   let title = isRelevantError ? 'Something went wrong...' : 'Something went wrong on our side...'
-  if (isValid) {
-    props.screenProps.pop({ isValid })
-  }
 
-  const gotoFR = () => {
-    props.screenProps.navigateTo('FaceVerification', { showHelper: true })
-  }
+  useEffect(() => {
+    if (isValid) {
+      screenProps.pop({ isValid })
+    }
+  }, [isValid])
+
+  const fullName = useMemo(() => {
+    const store = GDStore.useStore()
+    const { fullName } = store.get('profile')
+
+    return fullName
+  }, [])
+
+  const gotoFR = useCallback(() => {
+    screenProps.navigateTo('FaceVerification', { showHelper: true })
+  }, [screenProps])
 
-  log.debug(props.screenProps)
+  log.debug(screenProps)
```

Comment:
pls wrap to useEffect (as i explained this morning)

Reply:
done